splunk-otel-android/src/main/java/com/splunk/rum/FileUtils.java

```diff
@@ -0,0 +1,58 @@
+package com.splunk.rum;
+
+import static com.splunk.rum.SplunkRum.LOG_TAG;
+
+import android.util.Log;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+// Basic wrapper around filesystem operations, primarily for testing
+public class FileUtils {
+
+    void writeAsLines(Path tmpFile, List<byte[]> blocksOfData) throws IOException {
+        try (FileOutputStream out = new FileOutputStream(tmpFile.toFile())) {
```

Comment:
You can use `Files.newOutputStream()` instead of the legacy `File` API

Reply:
Impl changed to remove `Path` usages.